src/procedures/proc_procedures.c

```diff
@@ -0,0 +1,89 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "proc_procedures.h"
+#include "../util/arr.h"
+#include "rax.h"
+#include "procedure.h"
+
+// CALL dbms.procedures()
+
+typedef struct {
+	raxIterator *current_proc; // Current proc
+	SIValue *output; // Array with a maximum of 4 entries: ["name", name, "mode", mode].
+} ProceduresContext;
+
+ProcedureResult Proc_ProceduresInvoke(ProcedureCtx *ctx,
+  const SIValue *args, const char **yield) {
+	if(array_len((SIValue *)args) != 0) return PROCEDURE_ERR;
```

Comment:
why the casting?

Reply:
removing the casting gives a warning: "Passing 'const SIValue *' (aka 'const struct SIValue *') to parameter of type 'array_t' (aka 'void *') discards qualifier"